src/Runtime/Eval.hs

```diff
@@ -56,21 +52,14 @@ bindings vs = map (bind x) vs
     x = map (bind []) vs
 
 -- | Bind an individual valdef to its name in the current Enviroment
-bind :: Env -> ValDef -> (Name, V)
-bind env (Val _ (Veq n e)) = (n, Simple e env)
-bind env (Val _ (Feq n (Pars ls) e)) = (n, Fun ls e env)
+bind :: Env -> ValDef -> (Name, Val)
+bind env (Val _ (Veq n e)) = (n, v)
+  where
+    v = runIdentity (runReaderT (eval e) env)
+bind env (Val _ (Feq n (Pars ls) e)) = (n, Vf ls env e)
 -- bval?
 -- builtins
 
--- | Builtin input function (SUBJECT TO CHANGE)
-input :: [Expr] -> Eval Val
-input [] = (Vpos . read) <$> (liftIO getLine)
-input _ = undefined
-
--- | List of all the builtins in the builtin enviroment
-builtins :: [(Name, [Expr] -> Eval Val)]
-builtins = [("input", input)]
-
```

Comment:
What happened with this by the way?

Reply:
Input isn't a function currently in the syntax/example TicTacToe but a variable that is referenced. I think it should be a function, that takes the board from where you receive input. This will make it explicit which board should be displayed to the user and ask for input on the GUI (and makes it easier for us to implement that too, as it stands I think we'd have to just try and find the Board that's highest up in the scope and display that, which isn't very explicit).
the builtins (except input) I will implement on a different branch. They'll be functions `[Expr] -> Eval Val` still, I believe